### PR TITLE
Modals: Mobile - Get alignment right

### LINKED ISSUE #1593: Mobile: Modal top margin is too long
image

This is wrong for _all_ info modals.

separate from #1575

### PR DESCRIPTION
**Pull request by machikoyasuda**
fix #1593 

## What this PR does
- The informational modals have too much margin-top and margin-left and margin-right. This PR brings the text up by 36px so it's aligned with the X, and makes adjustments necessary for that.
- The overall effect of getting all the margins right is that modals on mobile fit more text on the screen, and theoretically, are shorter now.

## How to test this PR
- Run thru the app in mobile and open _every_ informational modal (2 login.gov, 1 contactless pay, 1 littlepay)

## Docs: Informational Modals, specs

I wrote these out to help me get the alignment right.

### Desktop
- The height of the entire header section: 36px
image

- Space between the edge and the begining of the text: 32px
image

### Mobile
- The distance between the edge and the top of the X/the begining of header text: 16px
image

- Space between the edge and the begining of the text: 16px
image

- I added the `z-index` declarations (and a `margin-right`) so that the X always comes on _top_ of the header and the header never overlaps with the X. Now that the header is getting pushed up 36px _into_ the header div, we need this z-index to ensure the clickability of the X.
image

### CODE REVIEW

**Comment by machikoyasuda** [comment]
Note: There's a change in the line-height of the modal titles on Mobile, which makes these 2-line titles tighter (vertically shorter) in Figma. But that's not part of this PR, that's part of #1573.

File: benefits/eligibility/templates/eligibility/includes/modal--contactless.html [hunk 1]
```diff
@@ -3,7 +3,7 @@
 {% load static %}
 
 {% block modal-content %}
-    <h2 id="contactless-modal">{% translate "eligibility.pages.start.modal.title" %}</h2>
+    <h2 id="contactless-modal" class="me-4 me-md-0">{% translate "eligibility.pages.start.modal.title" %}</h2>
```
Comment (thekaveman) [inline benefits/eligibility/templates/eligibility/includes/modal--contactless.html:6]: What are the `me-4` and `me-md-0` classes here? I couldn't find that prefix `me` in the [Bootstrap docs](https://getbootstrap.com/docs/4.1/utilities/spacing/).

Comment (machikoyasuda) [reply benefits/eligibility/templates/eligibility/includes/modal--contactless.html:6]: So the margin/padding class naming logic is like this for Bootstrap 5.0 and beyond:

- `m` or `p`: margin or padding
- `t`: top
- `b`: bottom
- `y`: top and bottom
- `x`: left and right

and here's the new, Left to Right/Right to Left language-compatible friendly version of:
- `s`: short for START. for our languages (English and Spanish), that means `left`
- `e`: short for END. for our languages (English and Spanish), that means `right`

Documented here: https://getbootstrap.com/docs/5.3/utilities/spacing/#notation

Comment (machikoyasuda) [reply benefits/eligibility/templates/eligibility/includes/modal--contactless.html:6]: Actually this is the real link for our version: https://getbootstrap.com/docs/5.1/utilities/spacing/

You were referencing 4.1, but this app uses 5.1

Comment (machikoyasuda) [reply benefits/eligibility/templates/eligibility/includes/modal--contactless.html:6]: TL;DR - `e` means `end`. As in the end of the line for LTR text, so it's `right`.

Comment (thekaveman) [reply benefits/eligibility/templates/eligibility/includes/modal--contactless.html:6]: Doh, my bad on the version mismatch. Thanks!

**Comment by thekaveman** [comment]
I think one of our paddings/margins needs to be adjust down to `sm`? 

There is a funny break between `767px` and `992px` where the modal looks mobile (i.e. no padding at the top), but the content looks tablet/desktop. 

image

**Review by thekaveman — COMMENTED** [review]
~I think this makes the Close `X` on the Agency Selector modal a little too close to the text for certain sizes.~

Sorry I though I did, but I hadn't updated my local branch with [587732b](https://github.com/cal-itp/benefits/pull/1599/commits/587732ba9797e334e77e8a7127e0219dc0142880)

**Review by thekaveman — APPROVED** [review]
This is great!

Testing note: if your local site is already running, and you pull in new changes, _some things will not rebuild_

* `styles.css` is fine if you hard refresh your browser, you'll get the updated version
* most templates will be updated
* I found in reviewing this PR that the `selection-label--senior.html` include, which is for the radio button on Eligibility Index, _was not updating_ -- I had to restart my local site to get the change. This may be because it is referenced through a widget for a form field, which maybe gets compiled a single time at the beginning?

**Pull request merged**